pkg/broker/eventutil/label.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventutil
+
+import (
+	"strings"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+const (
+	// LabelPrefix is the prefix for label keys.
+	LabelPrefix = "kgcp"
+)
+
+// LabeledEvent is a wrapper of a cloudevent
+// that allows labeling the event.
+type LabeledEvent struct {
+	event *cloudevents.Event
+}
+
+// NewLabeledEvent creates a new LabeledEvent.
+func NewLabeledEvent(e *cloudevents.Event) *LabeledEvent {
+	return &LabeledEvent{event: e}
+}
+
+// WithLabel attaches a label to the event as an extension.
+func (le *LabeledEvent) WithLabel(key, value string) *LabeledEvent {
+	le.event.SetExtension(LabelPrefix+key, value)
+	return le
+}
+
+// GetLabels gets all the labels as a map.
+func (le *LabeledEvent) GetLabels() map[string]string {
+	m := make(map[string]string)
+	exts := le.event.Extensions()
+	for k, v := range exts {
+		if strings.HasPrefix(strings.ToLower(k), LabelPrefix) {
+			m[k] = v.(string)
```

Comment:
Should this remove the prefix too? So that it is symmetric with `WithLabel`.

Reply:
Ah, I forgot to test this.